File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/content/DocumentAddBundleModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.content;

import ro.cloudSoft.cloudDoc.presentation.client.shared.model.bpm.WorkflowModel;
import ro.cloudSoft.cloudDoc.presentation.client.shared.model.bpm.WorkflowStateModel;

public class DocumentAddBundleModel {
	
	private DocumentTypeModel documentType;
	private WorkflowModel workflow;
	private WorkflowStateModel currentState;
	private Boolean canUserSend;
	
	public DocumentTypeModel getDocumentType() {
		return documentType;
	}
	public void setDocumentType(DocumentTypeModel documentType) {
		this.documentType = documentType;
	}
	public WorkflowModel getWorkflow() {
		return workflow;
	}
	public void setWorkflow(WorkflowModel workflow) {
		this.workflow = workflow;
	}
	public WorkflowStateModel getCurrentState() {
		return currentState;
	}
	public void setCurrentState(WorkflowStateModel currentState) {
		this.currentState = currentState;
	}	
	public void setCanUserSend(Boolean canSend) {
		this.canUserSend = canSend;
	}
	public Boolean getCanUserSend() {
		return canUserSend;
	}
}

File: CloudFlow/FrontEnd-App/src/app/shared/model/content/document-add-bundle.model.ts
import { JsonObject, JsonProperty } from "json2typescript";
import { DocumentTypeModel } from "./document-type.model";
import { WorkflowModel, WorkflowStateModel } from "./../bpm";

@JsonObject
export class DocumentAddBundleModel {

	@JsonProperty("documentType", DocumentTypeModel)
	public documentType: DocumentTypeModel = null;

	@JsonProperty("workflow", WorkflowModel, true)
	public workflow: WorkflowModel = null;

	@JsonProperty("currentState", WorkflowStateModel, true)
	public currentState: WorkflowStateModel = null;

	@JsonProperty("canUserSend", Boolean, true)
	public canUserSend: boolean = false;
}